pkg/util/provider/machinecontroller/machine_util.go

```diff
@@ -944,77 +944,94 @@ func (c *controller) setMachineTerminationStatus(ctx context.Context, deleteMach
 	return machineutils.ShortRetry, err
 }
 
-// getVMStatus tries to retrive VM status backed by machine
+// getVMStatus tries to retrieve VM status backed by machine
 func (c *controller) getVMStatus(ctx context.Context, getMachineStatusRequest *driver.GetMachineStatusRequest) (machineutils.RetryPeriod, error) {
 	var (
 		retry       machineutils.RetryPeriod
 		description string
 		state       v1alpha1.MachineState
+		err         error
 	)
-
-	statusResp, err := c.driver.GetMachineStatus(ctx, getMachineStatusRequest)
-	if err == nil {
-		// VM Found
-
-		// If `node` label is missing on machine obj, then update this label on Machine object with nodeName from status response
-		nodeName := getMachineStatusRequest.Machine.Labels[v1alpha1.NodeLabelKey]
-		if nodeName == "" {
-			err = c.updateMachineNodeLabel(ctx, getMachineStatusRequest.Machine, statusResp.NodeName)
-			if err != nil {
-				return machineutils.ShortRetry, err
+	// Node label is required for drain of node, therefore we try to update machine object before proceeding to drain.
+	isNodeLabelUpdated := false
+	//check if node name label is already present in machine object
+	nodeName := getMachineStatusRequest.Machine.Labels[v1alpha1.NodeLabelKey]
+	if isValidNodeName(nodeName) {
```

Comment:
this function is actually quite useless. It only checks if the node Name is empty. I would just remove this function.